x/reports/internal/keeper/keeper.go

```diff
@@ -0,0 +1,78 @@
+package keeper
+
+import (
+	"bytes"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/posts"
+	"github.com/desmos-labs/desmos/x/reports/internal/types"
+	"github.com/desmos-labs/desmos/x/reports/internal/types/models"
+)
+
+// Keeper maintains the link to data storage and exposes getter/setter methods for the various parts of the state machine
+type Keeper struct {
+	postKeeper posts.Keeper // post's keeper to perform checks on the postIDs
+	StoreKey   sdk.StoreKey // Unexposed key to access store from sdk.Context
+	Cdc        *codec.Codec // The wire codec for binary encoding/decoding.
+}
+
+// NewKeeper creates new instances of the reports Keeper
+func NewKeeper(pk posts.Keeper, cdc *codec.Codec, storeKey sdk.StoreKey) Keeper {
+	return Keeper{
+		postKeeper: pk,
+		StoreKey:   storeKey,
+		Cdc:        cdc,
+	}
+}
+
+// CheckPostExistence checks if a post with the given postID is present inside
+// the current context and returns a boolean indicating that.
+func (k Keeper) CheckPostExistence(ctx sdk.Context, postID posts.PostID) bool {
+	_, exist := k.postKeeper.GetPost(ctx, postID)
+	return exist
+}
+
+// SaveReport allows to save the given reports inside the current context.
+// It assumes that the given reports has already been validated.
+// If the same reports has already been inserted, nothing will be changed.
+func (k Keeper) SaveReport(ctx sdk.Context, postID posts.PostID, report types.Report) {
+	store := ctx.KVStore(k.StoreKey)
+	key := models.ReportStoreKey(postID)
+	// Get the list of reports related to the given postID
+	var reports models.Reports
```

Comment:
Better adding a space here IMO

```suggestion
	key := models.ReportStoreKey(postID)
	
	// Get the list of reports related to the given postID
	var reports models.Reports
```